games-fps/chocolate-doom/chocolate-doom-3.0.1.ebuild

```diff
@@ -0,0 +1,82 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{7,8} )
+
+inherit autotools python-any-r1 xdg
+
+DESCRIPTION="A Doom source port that is minimalist and historically accurate"
+HOMEPAGE="https://www.chocolate-doom.org"
+SRC_URI="https://github.com/${PN}/${PN}/archive/${P}.tar.gz"
+
+LICENSE="BSD GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="bash-completion doc libsamplerate +midi png vorbis"
+
+DEPEND="
+	media-libs/libsdl2
```

Comment:
This needs at least `[video]` added. It looks like `sound` and `joystick` support are optional.

Reply:
You are right, `video` at the very least should be set.

I was actually wondering whether there should be USE flags for `sound`, `joystick`, `midi`, and `vorbis`. Since these are just passed directly to `media-libs/libsdl2` and `media-libs/sdl2-mixer`, does it make sense to create them for the chocolate-doom ebuild? These features are all optional so I wonder if it's better to have users enable them in their respective ebuilds rather than passing them down from the chocolate-doom ebuild.

Reply:
It's a grey area. If it's strictly required or it makes a difference at build time then there should definitely be a USE flag. If it's a flag they're less likely to have enabled like `midi` then maybe. `sound` and `joystick` are enabled on libsdl2 by default so probably not.

Reply:
Yes, I think I'll leave `sound` and `joystick` out of this ebuild since few users will need them and they are on by default anyway. `midi` and `vorbis` for that matter don't affect chocolate-doom build time either, but their need is not as obvious to the users so I'll leave them in.